src/backend/utils/indexer.js

```diff
@@ -98,9 +98,9 @@ const search = async (textToSearch, filter = 'post') => {
   const {
     body: { hits },
   } = await client.search({
-    from: 0,
+    from: page * perPage,
```

Comment:
What if `page` is `undefined`?  We better deal with the case that it's not passed to this function and use `0`.  Also, what if you go past the end of the result list?  We need to add a check to make sure we don't request a page beyond the end of the results.

Reply:
set page to default to 0 if not passed the arg. I wrote the logic for checking if in the last page of the result in searchResults.jsx hope you can take a look since it can probably be improved 😂 its the only thing i could come up with.

Reply:
This stuff gets complicated, and I don't have a ton of experience tuning ES, so probably we should do the simple/naive thing for now, and fix it later in follow-ups.  There are extensive docs on this in https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html.

Based on that, what about something like:

```
from: calculateFrom(page, perPage),
size: perPage
```

where `calculateFrom` is something like:

```js
const calculateFrom = (page, perPage) => {
  const ES_MAX = 10000; // 10K is the upper limit of what ES will return without issue for searches
  const wanted = page * perPage;
  // Don't exceed 10K, and if we will, return an offset under it by one page size
  return wanted + perPage < ES_MAX ? wanted : ES_MAX - perPage
};
```

This code is 100% untested, but you can use it as a starting point :)

Reply:
This wasn't fixed.  Please don't resolve if you don't actually fix a review comment.

Reply:
sorry about that, the comment the previous comment didnt load until i refreshed the page

Reply:
I implemented the code that you wrote and tested it, it works correctly. Let me know if this is resolved for now.